data_source.go

```diff
@@ -89,13 +87,38 @@ func Start(ds DataSource) error {
 	return nil
 }
 
+// RowColCode holds an 8-byte summary of the row-column geometry
```

Comment:
I admit that this packing of row/col number and counts is probably overkill, but it was fun!